src/app/components/language-selector/language-selector.tsx

```diff
@@ -19,29 +26,24 @@ export function LanguageLink({locale, slug}) {
     return (<a {...props}><LanguageText locale={locale} /></a>);
 }
 
-// Provide a fallback so ancient browsers don't outright fail
-export function LanguageText({locale}) {
-    if (Intl.DisplayNames) {
-        return (<LanguageTextUsingIntl locale={locale} />);
-    }
-    return locale;
-}
-
-function LanguageTextUsingIntl({locale}) {
+export function LanguageText({locale}: {locale: string}) {
     const {language} = useLanguageContext();
-    const languageName = React.useMemo(
-        () => new Intl.DisplayNames([language], {type: 'language'}),
-        [language]
+    const text = React.useMemo(
+        () => (new Intl.DisplayNames([language], {type: 'language'})).of(locale),
```

Comment:
We don't support browsers so old that Intl.DisplayNames isn't supported, so I took the fallback out.